### PR TITLE
Replace the deprecated uses of expression closures with arrow functio…

### PR DESCRIPTION
**Pull request by Dexterp37**
…ns. Also changes for-of to Array.map where needed.

This is needed to make Thunderbird Conversations work with the latest Daily Thunderbird.

### CODE REVIEW

File: msgHdrUtils.js [hunk 1]
```diff
@@ -87,8 +87,9 @@ XPCOMUtils.defineLazyGetter(MailServices, "messenger", function () {
  * @param {nsIMsgDbHdr} aMsg The message
  * @return {String}
  */
-function msgHdrGetUri (aMsg)
+function msgHdrGetUri (aMsg) {
   aMsg.folder.getUriForMsg(aMsg)
```
Comment (msprotz) [inline msgHdrUtils.js]: missing return?

File: msgHdrUtils.js [hunk 2]
```diff
@@ -110,40 +111,45 @@ function msgUriToMsgHdr(aUri) {
  * @param {nsIMsgDbHdr} msgHdr The message header to examine
  * @return {bool}
  */
-function msgHdrIsInbox(msgHdr)
+function msgHdrIsInbox(msgHdr) {
   msgHdr.folder.getFlag(nsMsgFolderFlags_Inbox)
```
Comment (msprotz) [inline msgHdrUtils.js]: same

File: msgHdrUtils.js [hunk 3]
```diff
@@ -110,40 +111,45 @@ function msgUriToMsgHdr(aUri) {
...
+}
 
 /**
  * Tells if the message is a draft message
  * @param {nsIMsgDbHdr} msgHdr The message header to examine
  * @return {bool}
  */
-function msgHdrIsDraft(msgHdr)
+function msgHdrIsDraft(msgHdr) {
   msgHdr.folder.getFlag(nsMsgFolderFlags_Drafts)
```
Comment (msprotz) [inline msgHdrUtils.js]: same

File: msgHdrUtils.js [hunk 4]
```diff
@@ -110,40 +111,45 @@ function msgUriToMsgHdr(aUri) {
...
+}
 
 /**
  * Tells if the message is a sent message
  * @param {nsIMsgDbHdr} msgHdr The message header to examine
  * @return {bool}
  */
-function msgHdrIsSent(msgHdr)
+function msgHdrIsSent(msgHdr) {
   msgHdr.folder.getFlag(nsMsgFolderFlags_SentMail)
```
Comment (msprotz) [inline msgHdrUtils.js]: same

File: msgHdrUtils.js [hunk 5]
```diff
@@ -110,40 +111,45 @@ function msgUriToMsgHdr(aUri) {
...
+}
 
 /**
  * Tells if the message is an archived message
  * @param {nsIMsgDbHdr} msgHdr The message header to examine
  * @return {bool}
  */
-function msgHdrIsArchive(msgHdr)
+function msgHdrIsArchive(msgHdr) {
   msgHdr.folder.getFlag(nsMsgFolderFlags_Archive)
```
Comment (msprotz) [inline msgHdrUtils.js]: same

File: msgHdrUtils.js [hunk 6]
```diff
@@ -110,40 +111,45 @@ function msgUriToMsgHdr(aUri) {
...
+}
 
 /**
  * Get a nsIMsgDbHdr from a Necko URL.
  * @param {String} The URL
  * @return {nsIMsgDbHdr} The message header.
  */
-function msgHdrFromNeckoUrl(aUrl)
+function msgHdrFromNeckoUrl(aUrl) {
   aUrl.QueryInterface(Ci.nsIMsgMessageUrl).messageHeader
```
Comment (msprotz) [inline msgHdrUtils.js]: same

**Comment by msprotz** [comment]
Great pull request too, I'll let you fix things up then I'll merge. Thanks!!

**Comment by Dexterp37** [comment]
Thanks for the review :-)

**Comment by protz** [comment]
Thanks again!

**Pull request merged**